app/src/main/java/com/example/habitsmasher/ui/history/HabitEventListFragment.java

```diff
@@ -0,0 +1,294 @@
+package com.example.habitsmasher.ui.history;
+
+import android.content.Context;
+import android.net.Uri;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.DefaultItemAnimator;
+import androidx.recyclerview.widget.DividerItemDecoration;
+import androidx.recyclerview.widget.ItemTouchHelper;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.example.habitsmasher.Habit;
+import com.example.habitsmasher.HabitEvent;
+import com.example.habitsmasher.HabitEventList;
+import com.example.habitsmasher.R;
+import com.example.habitsmasher.User;
+import com.example.habitsmasher.ui.dashboard.HabitListFragment;
+import com.firebase.ui.firestore.FirestoreRecyclerOptions;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.Query;
+import com.google.firebase.storage.FirebaseStorage;
+import com.google.firebase.storage.StorageReference;
+import com.google.firebase.storage.UploadTask;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.UUID;
+
+/**
+ * HabitEventListFragment class
+ * Responsible for data and UI handling of the habit event list
+ */
+public class HabitEventListFragment extends Fragment {
```

Comment:
Same thing here, since we have two "ListFragments" now, can we extract away any commonalities between the two lists to an interface? Things like the swipe functionality are shared between the two.

Reply:
Similarly to the other comment, although I agree that this is a good idea, it will need further investigation. Even with things like the swipe functionality that is shared, they're doing slightly different things (one removes a habit event, the other a habit). Some reworking and talk will be needed which I think should be done separately of this PR

Reply:
Similar to my reply to your comment, I was just starting up a discussion here. 

I think there are clear similarities, and we now have two very similar list fragments and associated adapters. Again, since interfaces don't actually define the functionality, it doesn't matter **what** the swipe does or if it's different, we just need **some** swipe function in there, and the details can be defined later.

Essentially, this enforces OOP and builds a contract for any future classes we build, serving as a reminder that we need to implement a swipe function (for example).